### PR TITLE
ssh: update README for ssh-askpass

### PR DESCRIPTION
**Pull request by crivotz**
## Proposed Changes
Added _Requirements_ section for `ssh-askpass` on Linux machine

### CODE REVIEW

**Comment by indrajitr** [comment]
Thanks @crivotz!

Also wondering if we should have _guards_ around implicit calls to `ssh-askpass` so that we don't rely on users to read the _README_ explicitly.

Historically, prezto hasn't documented explicit requirement for a feature to work. It is meant to _just work&trade;_ if the requirements are available.

NB: I am not against explicitly documenting _Requirements_, but there isn't any right now (so this might be a _first_). And we don't want to end up being sloppy over time and deviate from _just works&trade;_ principal by relying more on users having to read _README_ to get started.

**Comment by crivotz** [comment]
Thanks @indrajitr 
I am agree with you. I thought a lot about the word 'requirements' and I looked in the other modules to find alternatives.
Without the package you have no error but the agent will not work properly and will continue to require. the passphrase.
I try to look deep in the ssh-askpass ;)

**Comment by crivotz** [comment]
I hope to be on the right track
I cleaned the README and only for linux machines if askpass is not present I use the old way.

**Comment by diraol** [comment]
Hi @crivotz ,

in order to reduce code lines and checks, I think that you can first create a "local variable" (`ssh_id_path`) for example, by checking if there is any `_ssh_identities` (if there isn't, this local variable could be just 'empty'.

Then you check the SO and check for ASKPASS and use this variable.

I'm not good at all with shell/zsh syntax, but it would be something like:

```shell
  if (( ${#_ssh_identities} > 0 )); then
    local_ssh_id = "$_ssh_dir/${^_ssh_identities[@]}"
  else
    local_ssh_id = ""
  fi
  # Check for Linux system and ssh-askpass presence
  if [[ $(uname -s) == Linux* ]] && [[ ! -a /etc/X11/app-defaults/SshAskpass ]]; then
    ssh-add "$local_ssh_id"  2> /dev/null
  else
    ssh-add "$local_ssh_id"  < /dev/null 2> /dev/null
  fi
```

Sorry if it does not make sense, just trying to help. ;)

File: modules/ssh/init.zsh [hunk 1]
```diff
@@ -40,9 +40,19 @@ fi
 if ssh-add -l 2>&1 | grep -q 'The agent has no identities'; then
   zstyle -a ':prezto:module:ssh:load' identities '_ssh_identities'
   if (( ${#_ssh_identities} > 0 )); then
-    ssh-add "$_ssh_dir/${^_ssh_identities[@]}"  < /dev/null 2> /dev/null
+    # Check for Linux system and ssh-askpass presence
+    if [[ $(uname -s) == Linux* ]] && [[ ! -a /etc/X11/app-defaults/SshAskpass ]]; then
```
Comment (indrajitr) [inline modules/ssh/init.zsh]: * In zsh world, we do `if [[ "$OSTYPE" == linux* ]]`
* `/etc/X11/app-defaults/SshAskpass` might be too distro-specific, could there be something more generic?

File: modules/ssh/init.zsh [hunk 2]
```diff
@@ -40,9 +40,19 @@ fi
...
+      ssh-add "$_ssh_dir/${^_ssh_identities[@]}"  2> /dev/null
```
Comment (indrajitr) [inline modules/ssh/init.zsh]: You can probably replace `if (( ${#_ssh_identities} > 0 ))` block by inlining the check with something like:
```
ssh-add "${_ssh_identities:+$_ssh_dir/${^_ssh_identities[@]}}"  < /dev/null 2> /dev/null
```
I haven't tested though :)

File: modules/ssh/init.zsh (hunk 1 above)
Comment (crivotz) [inline modules/ssh/init.zsh]: - Ok I fix with $OSTYPE
- I can check for `/usr/bin/ssh-askpass` (symbolic link to `/etc/alternatives/ssh-askpass`) or for `/usr/lib/ssh/x11-ssh-askpass`

File: modules/ssh/init.zsh (hunk 2 above)
Comment (crivotz) [inline modules/ssh/init.zsh]: Work fine ;)

**Comment by crivotz** [comment]
Changes made

**Comment by indrajitr** [comment]
Looks good, thanks! Would you please squash the changes into a single commit?

**Comment by crivotz** [comment]
@indrajitr done

**Comment by indrajitr** [comment]
> ssh: update README for ssh-askpass

The commit message might have to be adjusted a bit. For example, the first line mentions updating README, which isn't the case anymore :)

**Comment by crivotz** [comment]
Sorry I left the standard message from the squash just for history.
I fix with a single and clear commit

**Comment by indrajitr** [comment]
Thank you very much @crivotz for your contribution and for dealing with these review requests patiently :)

**Comment by crivotz** [comment]
Thanks @indrajitr for your advice and for your help

**Pull request merged**